src/deck.py

```diff
@@ -1,27 +1,35 @@
 from card import Card
 import random
+from dataclasses import dataclass, field
+from functools import partial
+
 SUITS = ["C", "D", "H", "S"]
 RANKS = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K", "A"]
+CARDS = [Card(rank=r, suit=s) for r in RANKS for s in SUITS]
+
 
-#TODO: implement ranking
+# TODO: implement ranking
+@dataclass
 class Deck:
-    def __init__(self):
-        self.cards = []
-        for suit in (SUITS):
-            for rank in (RANKS):
-                card = Card(rank, suit)
-                self.cards.append(card.to_json)
-                
-        random.shuffle(self.cards)
-    
-    def pop(self):
-        return self.cards.pop()
-    
-    def draw(self, num):
-        cards_drawn = []
-        #if(self.cards.__sizeof__ < num):
-            #num = self.cards.__sizeof__
-        for _ in range(num):
-            cards_drawn.append(self.pop())
-        return cards_drawn
+    cards: list[Card] = field(
+        default_factory=partial(random.sample, CARDS, len(CARDS))
+    )
+    top: int = 0
+
+    def pop(self) -> dict[str, str]:
```

Comment:
I mentioned this in the previous PR, but it seems like `pop` should be an implementation detail. I don't understand what it means to "pop" a deck.

Reply:
@semmypurewal I think it makes sense, to some extent, as it's comparable to `list.pop`. But the main reason I left it is to preserve most of `Deck`'s API.

Reply:
Totally fine, it's just when abstracting a concept as a "class" it's usually good to only expose methods that make sense for the concept.